File: 02_hash_maps/d08_destination_city.go
// https://leetcode.com/problems/destination-city/description/

package main

import "fmt"

func destCity(paths [][]string) string {
	sources := make(map[string]bool)
	for _, v := range paths {
		sources[v[0]] = true
	}
	for _, v := range paths {
		if _, ok := sources[v[1]]; !ok {
			return v[1]
		}
	}
	return ""
}

func main() {
	testPaths := [][]string{{"London", "New York"}, {"New York", "Lima"}, {"Lima", "Sao Paulo"}}
	fmt.Println(destCity(testPaths))
}

File: 02_hash_maps/d08_destination_city.py
# https://leetcode.com/problems/destination-city/description/

# set solution
def solution(paths):
    sources = set()
    for path in paths:
        sources.add(path[0])
    for path in paths:
        if path[1] not in sources:
            return path[1]
        

paths = [["London","New York"],["New York","Lima"],["Lima","Sao Paulo"]]
print(solution(paths))
